src/Grpc.Net.ClientFactory/GrpcClientServiceExtensions.cs

```diff
@@ -306,21 +306,23 @@ public static class GrpcClientServiceExtensions
     {
         ArgumentNullThrowHelper.ThrowIfNull(services);
 
-        services
-            .AddHttpClient(name)
-            .ConfigurePrimaryHttpMessageHandler(() =>
-            {
-                // Set PrimaryHandler to null so we can track whether the user
-                // set a value or not. If they didn't set their own handler then
-                // one will be created by PostConfigure.
-                return null!;
-            });
+        services.AddHttpClient(name);
+
+        // Get PrimaryHandler and store it to compare later. Used to track whether the user set a handler or not.
+        // This action comes before registered user actions so the primary handler here will be the one created by the factory.
+        HttpMessageHandler? initialPrimaryHandler = null;
+        services.Configure<HttpClientFactoryOptions>(name, options =>
+        {
+            options.HttpMessageHandlerBuilderActions.Add(b => initialPrimaryHandler = b.PrimaryHandler);
+        });
 
         services.PostConfigure<HttpClientFactoryOptions>(name, options =>
         {
             options.HttpMessageHandlerBuilderActions.Add(builder =>
             {
-                if (builder.PrimaryHandler == null)
+                // If the primary handler is unchanged from what the factory created then replace the handler
+                // with one that has settings optimized for a gRPC client.
+                if (builder.PrimaryHandler == initialPrimaryHandler)
```

Comment:
> If the primary handler is unchanged from what the factory created

It is possible to modify the existing instance, so theoretically the user could have made some changes to a "factory-created" handler -- that would be lost here 🤔

While I assume it never happened with the existing code, bc currently there was always null instead of a "factory-created" handler -- but it could happen in the future, and the loss of user configuration will be a bug.

imagine e.g.:

```c#
services.AddGrpcHttpClient<TClient>("name")
    .ConfigurePrimaryHttpMessageHandler((h, _) =>
	{
	    if (h is HttpClientHandler hch)
	    {
	        hch.Credentials = ...
	    }
	    else if (h is SocketsHttpHandler shh)
	    {
	        shh.Credentials = ...
	    }
	});
```

This will not work because the handler will be overwritten by the post-configure 🤔

Reply:
Hmm, that's true.

It feels more and more like the best thing to do is use `ConfigureHttpClientDefaults`. If `IHttpClientBuilder.Name` matches a gRPC client name, then set the primary handler at the start of configuration instead of the end.